java/arcs/core/util/Scheduler.kt

```diff
@@ -147,7 +148,8 @@ class Scheduler(
         val timeoutHandler = { throwable: Throwable ->
             if (throwable is TimeoutCancellationException) {
                 // TODO(b/160251910): Make logging detail more cleanly conditional.
-                log.debug(throwable) { "Scheduled tasks timed out." }
+                log.debug(throwable) { "Scheduled tasks timed out " +
+                    "[arcId=${scope.coroutineContext[CoroutineName]?.name}]." }
```

Comment:
I can't find any existing code retrieving arc id this way, so if there's a better way to do this let me know. Maybe also encapsulation-wise I should just log `coroutineName=xxx` rather than calling it `arcId`? Like, is the `Scheduler` arc/arc-id-aware? Did not occur to me until reading just now.

I ran tests through this and it appears to do the right thing. Since it's testing, some show null, but I also see:

```
[arcId=ArcId::arcs.core.host.TestingHost@1803669141-!246138207570178:arc]
[arcId=ArcId::test]
[arcId=ArcId::!246138207570178:arc]
[arcId=ArcId::allocator]
[arcId=null]
```

Reply:
This is actually a little redundant, since the coroutine context name is (optionally) in the log content itself. An example:

`00002 (Scheduler-Thread#0 @ArcId::test#1) Debug: Scheduler: executeAgenda [arcId=ArcId::test].`

Should we still do this?

Reply:
if it's already there, I don't think it's needed.

Reply:
Sounds good, it was a useful coroutine context learning experience for me in any case.